docs/src/man/joins1.md

```diff
@@ -0,0 +1,609 @@
+   5 │ marshmallow             0.03  caramel                  0.01        0.04 ⋯
+   6 │ marshmallow             0.03  chocolate sauce          0.05        0.08
+```
+
+## Multiple keys to join on
+
+Now that we have better chocolate, let’s learn how to `join` on multiple columns. Extending the 
+above joins to work with `2 keys` is very easy. In fact, all we have to do is pass a vector of 
+symbols or strings to the `on` argument of the `join` functions. To demonstrate this, let’s copy 
+and add another column to both of our datasets. This will contain city names where the users live.
+
+```jldoctest dataframe
+julia> C_names= copy(persons) # It make sure we don't mess with the original table
+5×2 DataFrame
+ Row │ id     name
+     │ Int64  String
+─────┼───────────────
+   1 │     1  Rohit
+   2 │     2  Mohit
+   3 │     3  Rahul
+   4 │     4  Vijay
+   5 │     5  Akshat
+
+julia> C_names.city = ["Orai", "Gwalior"][mod1.(1:5, 2)]
+5-element Vector{String}:
+ "Orai"
+ "Gwalior"
+ "Orai"
+ "Gwalior"
+ "Orai"
+
+julia> D_earnings = copy(earnings) # do the same for earnings
+6×2 DataFrame
+ Row │ id     salary
+     │ Int64  Int64
+─────┼───────────────
+   1 │     3    5000
+   2 │     4    1000
+   3 │     5    5000
+   4 │     6    8000
+   5 │     7    3000
+   6 │     8    9000
+
+julia> D_earnings.city = ["Orai", "Gwalior"][mod1.(1:6, 2)]
+6-element Vector{String}:
+ "Orai"
+ "Gwalior"
+ "Orai"
+ "Gwalior"
+ "Orai"
+ "Gwalior"
+
+julia> C_names
+5×3 DataFrame
+ Row │ id     name    city
+     │ Int64  String  String
+─────┼────────────────────────
+   1 │     1  Rohit   Orai
+   2 │     2  Mohit   Gwalior
+   3 │     3  Rahul   Orai
+   4 │     4  Vijay   Gwalior
+   5 │     5  Akshat  Orai
+
+julia> D_earnings
+6×3 DataFrame
+ Row │ id     salary  city
+     │ Int64  Int64   String
+─────┼────────────────────────
+   1 │     3    5000  Orai
+   2 │     4    1000  Gwalior
+   3 │     5    5000  Orai
+   4 │     6    8000  Gwalior
+   5 │     7    3000  Orai
+   6 │     8    9000  Gwalior
+```
+
+One way we can think of this is that we have 2 separate databases. One in New York and one in London. 
+With the systems not knowing about each other, they keep track of the users’ ids separately. So the 
+name of user 1 in London is not the same as user 1 in New York. Indeed they are different users! So 
+when we merge the 2 tables we want to make sure that the names and earnings are not only matched in 
+user ids but also on the database name. Let’s do some joining on both columns then:
+
+```jldoctest dataframe
+julia> innerjoin(C_names, D_earnings, on = [:id, :city])
+3×4 DataFrame
+ Row │ id     name    city     salary
+     │ Int64  String  String   Int64
+─────┼────────────────────────────────
+   1 │     3  Rahul   Orai       5000
+   2 │     4  Vijay   Gwalior    1000
+   3 │     5  Akshat  Orai       5000
+```
+
+As we can see, only Daniel appears in both datasets and lives in the same place.
```

Comment:
what is Daniel?

Reply:
I see you are using https://towardsdatascience.com/joining-dataframes-in-julia-c435e3da32f3. Have you read https://policy.medium.com/medium-terms-of-service-9db0094a1e0f?

Reply:
I stop reviewing here as we should first ask @niczky12 for permission to reuse.

Reply:
I am happy for this to be reused here. Glad my articles are useful somewhere. 😀

Reply:
Thank you for giving the permission. That is the spirit I love.

Reply:
If I can, I would like to ask for a reference to the medium article in comments if possible.

Reply:
This is exactly what I suggested @RohitRathore1 on Slack.